hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1884,57 +1899,120 @@ public OmKeyInfo lookupFile(OmKeyArgs args, String clientAddress)
         startKey = OzoneFSUtils.addTrailingSlashIfNeeded(keyName);
       }
 
+      int countEntries = 0;
+      Table keyTable = metadataManager.getKeyTable();
+      Iterator<Map.Entry<CacheKey<String>, CacheValue<OmKeyInfo>>>
+          cacheIter = keyTable.cacheIterator();
+      String startCacheKey = OZONE_URI_DELIMITER + volumeName +
+          OZONE_URI_DELIMITER + bucketName + OZONE_URI_DELIMITER +
+          ((startKey.equals(OZONE_URI_DELIMITER)) ? "" : startKey);
+      // Note: eliminating the case where startCacheKey could end with '//'
+
+      // First, find key in TableCache
+      while (cacheIter.hasNext() && numEntries - countEntries > 0) {
```

Comment:
We should get all entries in the cache. Otherwise continuation part will not work. 
As we want to merge all the result in cache and entries with DB.

For more Info read code comments in OmMetadataManagerImpl.java L723